### PR TITLE
SampleGen: csproj for Sample package

### PR DESCRIPTION
**Pull request by hzyi-google**
- Generates a `.csproj` file for `Google.Cloud.Api.V1.Samples` package, which references the gapic project and `CommandLine` package.

### CODE REVIEW

**Comment by hzyi-google** [comment]
@chrisdunelm @googleapis/samplegen 

I wasn't able to build the samples project emitted by artman directly due to the following error message: 

`CSC : error CS2017: Cannot specify /main if building a module or library`

However, when I copy the entire `Samples` directory to a local clone of `google-cloud-dotnet`, everything works fine.

Is there anything else done during the packaging process?

**Comment by hzyi-google** [comment]
@munkhuushmgl

File: src/main/java/com/google/api/codegen/transformer/csharp/CSharpBasicPackageTransformer.java [hunk 1]
```diff
@@ -41,6 +41,7 @@
       "csharp/gapic_snippets_csproj.snip";
   private static final String UNITTEST_CSPROJ_TEMPLATE_FILENAME =
       "csharp/gapic_unittest_csproj.snip";
+  private static final String SAMPLE_CSPROJ_TEMPLATE_FILENAME = "csharp/gapic_samples_csproj.snip";
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/transformer/csharp/CSharpBasicPackageTransformer.java]: nit: for consistency, have this line and the previous formatted similarly

**Comment by hzyi-google** [comment]
So C# generates project files [here](https://github.com/googleapis/google-cloud-dotnet/blob/abbbf43d00bec4335f091c497a47ca4ee252a675/tools/Google.Cloud.Tools.ProjectGenerator/Program.cs#L429), and therefore this generated file will at most be used for our own testing. I'll merge it and submit a separate PR to google-cloud-dotnet that will actually be generating the csproj file for Sample package.

File: src/main/java/com/google/api/codegen/transformer/csharp/CSharpBasicPackageTransformer.java (hunk 1 above)
Comment (hzyi-google) [reply src/main/java/com/google/api/codegen/transformer/csharp/CSharpBasicPackageTransformer.java]: Nope, formatter doesn't allow me to do that (so annoying!)

**Pull request merged**